tests/test_vsstorage.py

```diff
@@ -410,13 +413,19 @@ def test_files(self):
 
         s.sendAuthorized = MagicMock(side_effect=[self.MockedResponse(200, self.test_list_doc),
                                                   self.MockedResponse(200, self.test_list_doc_end)])
-
-        files_list = map(lambda f: f, s.files())
+        files_list = [f for f in s.files()]
         self.assertEqual(len(files_list),10)
-        s.sendAuthorized.assert_has_calls([
-            call('GET', '/API/storage/INVALIDNAME/file;start=0;includeItem=True;number=100?path=%2F', None, {'Accept': 'application/xml'}, rawData=False),
-            call('GET', '/API/storage/INVALIDNAME/file;start=10;includeItem=True;number=100?path=%2F', None, {'Accept': 'application/xml'}, rawData=False)
-        ])
+        arg1, arg2, arg3, arg4 = s.sendAuthorized.call_args[0]
+        self.assertEqual(arg1, 'GET')
+        self.assertEqual(arg3, None)
+        self.assertEqual(arg4, {'Accept': 'application/xml'})
+        parsed_url = urlparse(arg2)
+        self.assertEqual(parsed_url.path, '/API/storage/INVALIDNAME/file')
+        self.assertIn('start=10', parsed_url.params)
+        self.assertIn('includeItem=True', parsed_url.params)
+        self.assertIn('number=100', parsed_url.params)
+        query_dict = parse_qs(parsed_url.query)
+        self.assertEqual(query_dict['path'], ['/'])
```

Comment:
you're not testing the second call that is there in the master branch on line 418?

Reply:
Actually I think it the line above that is not getting tested for.

I think when I wrote this the thinking behind it was that for the second call to be there the first call must have worked.

If you like I can rewrite this so the first call is also tested. When building this I thought that was pointless.

Reply:
should probably put it in to make sure that the code is not skipping calls

Reply:
I worked out the syntax to do this now and added it.